backend/test/test_github_client.py

```
@@ -0,0 +1,118 @@
+	method, url, payload, headers = mock_client.last_call
+	assert method == "post"
+	assert url == "https://github.com/login/oauth/access_token"
+	assert headers == {"Accept": "application/json"}
+	assert payload["code"] == "code-xyz"
+
+
+@pytest.mark.asyncio
+async def test_get_github_access_token_raises_on_http_error(monkeypatch):
+	resp = DummyResponse(None, exc=Exception("status error"))
+	mock_client = AsyncClientMock(resp)
+	monkeypatch.setattr(httpx, "AsyncClient", lambda *a, **k: mock_client)
```

Ruff (0.15.11)

[warning] Unused lambda argument: `a`

(ARG005)